src/modules/database/database.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+/* eslint-disable no-useless-constructor */
+/* eslint-disable no-console */
+import mongoose from 'mongoose';
+
+class Database {
+  private static instace: Database | null = null;
+
+  private mongodb = mongoose;
+
+  private uri = 'mongodb://127.0.0.1:27017/local';
+
+  private constructor() {}
+
+  public static getInstance(): Database {
+    if (Database.instace === null) Database.instace = new Database();
+
+    return Database.instace;
+  }
+
+  public async connect(config =
+  { useNewUrlParser: true, useUnifiedTopology: true }): Promise<void> {
+    await this.mongodb.connect(this.uri, config,
+      (err) => {
+        if (err) {
+          console.error(err.message);
+        } else {
+          console.log('Successfully connected to mongodb');
+        }
+      });
+  }
+
+  public db(): typeof mongoose {
```

Comment:
Why you variable `db` is private if you can acces it by this method ?

Reply:
Thanks for the advise, I changed the variable to public and removed this `db` method.

Reply:
Do you really need to access this `db` directly?

Reply:
How about implementing a method that access this `db` internally ?

Reply:
It's a better ideia, I did it in the last commit